impeller/display_list/skia_conversions_unittests.cc

```diff
@@ -23,5 +25,142 @@ TEST(SkiaConversionsTest, ToColor) {
   ASSERT_TRUE(ScalarNearlyEqual(converted_color.blue, 0x20 * (1.0f / 255)));
 }
 
+TEST(SkiaConversionsTest, GradientStopConversion) {
+  // Typical gradient.
+  {
+    std::vector<flutter::DlColor> colors = {flutter::DlColor::kBlue(),
+                                            flutter::DlColor::kRed(),
+                                            flutter::DlColor::kGreen()};
+    std::vector<float> stops = {0.0, 0.5, 1.0};
+    const auto gradient =
+        flutter::DlColorSource::MakeLinear(SkPoint::Make(0, 0),          //
+                                           SkPoint::Make(1.0, 1.0),      //
+                                           3,                            //
+                                           colors.data(),                //
+                                           stops.data(),                 //
+                                           flutter::DlTileMode::kClamp,  //
+                                           nullptr                       //
+        );
+
+    std::vector<Color> converted_colors;
+    std::vector<Scalar> converted_stops;
+    skia_conversions::ConvertStops(gradient.get(), converted_colors,
+                                   converted_stops);
+
+    ASSERT_TRUE(ScalarNearlyEqual(converted_stops[0], 0.0f));
+    ASSERT_TRUE(ScalarNearlyEqual(converted_stops[1], 0.5f));
+    ASSERT_TRUE(ScalarNearlyEqual(converted_stops[2], 1.0f));
+  }
+
+  // Missing 0.0 first stop gradient.
```

Comment:
Move these tests into separate TEST clauses, i.e.

- ValidGradientsAreUnaffected
- MissingFirstStopImplicitlyIsAdded
- MissingLastStopImplicitlyIsAdded
- ClampValuesGreaterThan1_0
- ClampValuesLessThan0_0
- ...

Not only will (hypothetically) the tests run in parallel, but if there is a failure/refactor, we'll understand what broke. It also makes the tests match the docstring more easily.

Reply:
Done